staging/src/k8s.io/apimachinery/pkg/conversion/converter.go

```diff
@@ -355,16 +386,32 @@ func verifyConversionFunctionSignature(ft reflect.Type) error {
 //                 // conversion logic...
 //                 return nil
 //          })
+// DEPRECATED: Will be removed in favor of RegisterGenericConversionFunc
 func (c *Converter) RegisterConversionFunc(conversionFunc interface{}) error {
 	return c.conversionFuncs.Add(conversionFunc)
 }
 
 // Similar to RegisterConversionFunc, but registers conversion function that were
 // automatically generated.
+// DEPRECATED: Will be removed in favor of RegisterGenericConversionFunc
 func (c *Converter) RegisterGeneratedConversionFunc(conversionFunc interface{}) error {
 	return c.generatedConversionFuncs.Add(conversionFunc)
 }
 
+// RegisterUntypedConversionFunc registers a function that converts between a and b by passing objects of those
+// types to the provided function. The function *must* accept objects of a and b - this machinery will not enforce
+// any other guarantee.
+func (c *Converter) RegisterUntypedConversionFunc(a, b interface{}, fn ConversionFunc) error {
+	return c.generatedConversionFuncs.AddUntyped(a, b, fn)
+}
+
+// RegisterGeneratedUntypedConversionFunc registers a function that converts between a and b by passing objects of those
+// types to the provided function. The function *must* accept objects of a and b - this machinery will not enforce
+// any other guarantee.
+func (c *Converter) RegisterGeneratedUntypedConversionFunc(a, b interface{}, fn ConversionFunc) error {
```

Comment:
do we really need two register funcs? They do the same.

Reply:
a typo? The upper one should add to c.conversionFuncs, no?

Reply:
Upper one should have been conversionFuncs, fixed